test_conformance/images/kernel_read_write/test_common.cpp

```diff
@@ -17,3 +33,1514 @@ cl_sampler create_sampler(cl_context context, image_sampler_data *sdata, bool te
     return sampler;
 }
 
+void InitFloatCoordsCommon(image_descriptor *imageInfo,
+                           image_sampler_data *imageSampler, float *xOffsets,
+                           float *yOffsets, float *zOffsets, float xfract,
+                           float yfract, float zfract, int normalized_coords,
+                           MTdata d, int lod)
+{
+    size_t i = 0;
+    if (gDisableOffsets)
+    {
+        for (size_t z = 0; z < imageInfo->depth; z++)
```

Comment:
This requires callers to guarantee depth or height are never zero; is that the intent? I know this has bitten us in other places.

Reply:
I've modified this in later patches to account for all image types. In this change the code has only been moved so this is still only run for 3D images.